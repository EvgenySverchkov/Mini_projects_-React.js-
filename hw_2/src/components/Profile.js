import React from 'react';

const Profile = ({heandler}) => (
	
		<form style={wrapStyle} className="form-inline" onSubmit = {heandler}>
			<div className="form-group mx-sm-3 mb-2">
				<input className="form-control" id="userId" type="text" min="0" max="20" placeholder="Enter user's id"/>
			</div>
			<input type="submit" className="btn btn-primary mb-2" value="Submit"/>
		</form>

);

const wrapStyle = {
	width: '30%',
	margin: '0 auto'
}
export default Profile;