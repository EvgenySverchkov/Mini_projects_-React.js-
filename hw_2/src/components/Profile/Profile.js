import React from 'react';
import PropTypes from 'prop-types';

const Profile = ({heandler}) => (
	<form style={wrapStyle} className="form-inline" onSubmit = {heandler}>
		<div className="form-group mx-sm-3 mb-2">
			<input className="form-control" id="userId" type="number" placeholder="Enter user's id"/>
		</div>
		<input type="submit" className="btn btn-primary mb-2" value="Submit"/>
	</form>
);

const wrapStyle = {
	width: '30%',
	margin: '0 auto'
}

Profile.propTypes = {
	heandler: PropTypes.func
}
export default Profile;