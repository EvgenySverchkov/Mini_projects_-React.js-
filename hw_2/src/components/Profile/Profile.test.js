import React from 'react';
import testRenderer from 'react-test-renderer';

import Profile from './Profile';

it('Snapshot test for <Profile />', ()=>{
	const ProfileComponentTree = testRenderer.create(<Profile />).toJSON();
	expect(ProfileComponentTree).toMatchSnapshot();
});