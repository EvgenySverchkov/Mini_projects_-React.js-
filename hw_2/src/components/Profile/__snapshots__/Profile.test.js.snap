// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for <Profile /> 1`] = `
<form
  className="form-inline"
  style={
    Object {
      "margin": "0 auto",
      "width": "30%",
    }
  }
>
  <div
    className="form-group mx-sm-3 mb-2"
  >
    <input
      className="form-control"
      id="userId"
      placeholder="Enter user's id"
      type="number"
    />
  </div>
  <input
    className="btn btn-primary mb-2"
    type="submit"
    value="Submit"
  />
</form>
`;
